docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+
+## Challenge
+
+When we talk about Fybrik data plane isolation, we are talking about connectivity between a user workload, which is out of
+Fybrik scope, and Fybrik modules, which can be developed by third parties. Therefore, in this document (and in its
+extensions) we provide our recommendations how to protect data plane access and possible solutions. Responsibility of the
+data plane isolation is shared between Fybrik core, Fybrik users, module developers and probably IT teams.
+
+- Fybrik users should provide all relevant information about possible workload locations to Fybrik, correctly 
+configure their workloads. For example, to set labels on the workload and inform Fybrik about the labels and namespace 
+where the workload runs.
+
+- Depends on the isolation method, Fybrik should propagate all relevant information to the deployed modules, and/or
+correctly configure a chosen protection mechanism, e.g. Network Policies..
+
+- Module developers should allow to protect the module instances. For example, to allow TLS support 
+ 
+- In some complicated cases, IT teams interfere can be required, e.g. to configure external access points, firewalls and so on.
+
+## Requirements
+
+- Only predefined users/workloads should be able to access the data plane.
+- If a data plane consist of a module chain we should prevent authorized access to the intermediate modules too.
+- Modules should not be able to "illegally" communicate with modules outside of the defined data plane(s) and should 
+connect to only predefined data sets.
+- Currently, Fybrik assumes co-location of client workloads with data plane entry point modules. Future implementations
+might support deployment workloads on different clusters or run them out of Kubernetes as a standalone applications.
+  - We check the co-location of a Fybrik module and a relevant workload in the same cluster as a specific use case.
+- We should allow workload client connections isolation based on IP addresses. It can be useful if a workload is not 
+co-locate with a Fybrik data plane.  
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). NP can restrict incoming (ingress) 
+and/or outgoing (egress) communications. 
+The restrictions are based on combinations of Pod labels, Namespaces labels and IP blocks.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- a single Kubernetes cluster oriented, cross clusters restrictions can be based on IP blocks only, which significantly 
+complicates the configuration. Therefore, later we will provide other isolation methods as an extension to NP.
```

Comment:
Will we be able to support module chaining when modules are deployed on different clusters? Or does this mean that isolation will be limited in this case?

Reply:
I am not sure that I understand your question. We are going to present cross cluster connectivity, which can be done with different methods. In some cases, NP based isolation can be limited to IP blocks, which is not very convenient.